src/Microsoft.DotNet.Build.Tasks.Feed/src/AssetPublisherFactory.cs

```diff
@@ -38,22 +40,36 @@ public virtual IAssetPublisher CreateAssetPublisher(TargetFeedConfig feedConfig,
                     {
                         return new AzureStorageContainerAssetTokenCredentialPublisher(
                             new Uri(feedConfig.TargetURL),
-                            new DefaultAzureCredential(
-                                new DefaultAzureCredentialOptions
-                                {
-                                    ExcludeVisualStudioCodeCredential = true,
-                                    ExcludeVisualStudioCredential = true,
-                                    ExcludeAzureDeveloperCliCredential = true,
-                                    ExcludeInteractiveBrowserCredential = true,
-                                    ManagedIdentityClientId = task.ManagedIdentityClientId,
-                                    CredentialProcessTimeout = TimeSpan.FromSeconds(60.0)
-                                }),
+                            GetAzureTokenCredential(task.ManagedIdentityClientId),
                             _log);
                     }
                 default:
                     throw new NotImplementedException();
             }
         }
+
+        private ConcurrentDictionary<string, TokenCredential> _tokenCredentialsPerManagedIdentity = new ConcurrentDictionary<string, TokenCredential>(-1, 10);
+
+        private TokenCredential GetAzureTokenCredential(string managedIdentityClientId)
+        {
+            TokenCredential tokenCredential = _tokenCredentialsPerManagedIdentity.GetOrAdd(managedIdentityClientId, static (mi) =>
+            {
+                return new TokenCredentialShortCache(
+                    new DefaultAzureCredential(
+                        new DefaultAzureCredentialOptions
+                        {
+                            ExcludeVisualStudioCodeCredential = true,
+                            ExcludeVisualStudioCredential = true,
+                            ExcludeAzureDeveloperCliCredential = true,
+                            ExcludeInteractiveBrowserCredential = true,
+                            ManagedIdentityClientId = mi,
+                            CredentialProcessTimeout = TimeSpan.FromSeconds(60.0)
+                        }
+                    )
+                );
+            });
+            return tokenCredential;
```

Comment:
Could be this as well:

```suggestion
            return _tokenCredentialsPerManagedIdentity.GetOrAdd(managedIdentityClientId, static (mi) =>
                new TokenCredentialShortCache(
                    new DefaultAzureCredential(
                        new DefaultAzureCredentialOptions
                        {
                            ExcludeVisualStudioCodeCredential = true,
                            ExcludeVisualStudioCredential = true,
                            ExcludeAzureDeveloperCliCredential = true,
                            ExcludeInteractiveBrowserCredential = true,
                            ManagedIdentityClientId = mi,
                            CredentialProcessTimeout = TimeSpan.FromSeconds(60.0)
                        }));
```

Reply:
I will update it to this